server/src/main/kotlin/fi/vauhtijuoksu/vauhtijuoksuapi/server/impl/StreamMetadataRouter.kt

```diff
@@ -15,14 +18,20 @@ import io.vertx.ext.web.Router
 import io.vertx.ext.web.handler.AuthenticationHandler
 import io.vertx.ext.web.handler.BodyHandler
 import io.vertx.ext.web.handler.CorsHandler
+import mu.KotlinLogging
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+import java.util.Date
+import java.util.UUID
 import javax.inject.Inject
 import javax.inject.Named
 
 // Throwing as soon as some validation fails is the simplest way
 @Suppress("ThrowsCount")
 class StreamMetadataRouter
 @Inject constructor(
-    private val db: SingletonDatabase<StreamMetadata>,
+    private val dbStream: SingletonDatabase<StreamMetadata>,
```

Comment:
There's quite a lot of logic in this class now. Maybe add a separate service layer to handle everything not directly related to API

Reply:
If you mean breaking the router to components like game data I can give it a try but seems like a lot of work for little benefit

Reply:
Nah, I meant separating the logic of parsing input from APIs and such, from the logic of fetching and parsing data from the databases.

Reply:
Not mandatory right now, but something I think I'll implement in the future for endpoints with more logic than fetching from one DB and returning the results. Makes testing easier

Reply:
I think I now see what you mean. I'd suggest that make a seperate issue out of it to have uniform solution instead of me jerry rigging some unique solution here

Reply:
Alrighty, https://trello.com/c/h6TRQDVF/74-create-a-service-layer-between-databases-and-routers-to-separate-api-related-code-from-data-transformations-etc